addons/calendar/models/ir_http.py

```diff
@@ -28,3 +29,23 @@ def _auth_method_calendar(cls):
             raise BadRequest(error_message)
 
         cls._auth_method_public()
+
+    @classmethod
+    def _auth_method_calendar_user(cls):
+        _, args = cls._match(request.httprequest.path) # FIXME DRY _dispatch
+        login = args.get('login')
+        token = args.get('token')
+        if cls.authenticate_token(login, token):
+            return True
+
+        raise Unauthorized()
+
+    @classmethod
+    def authenticate_token(cls, login, token):
```

Comment:
Maybe this should live in odoo.http.OpenERPSession with the other authenticate() function, but maybe not since we aren't creating a session.